### PR TITLE
[CALCITE-3171] SemiJoin on conditions push down throws IndexOutOfBoun…

### PR DESCRIPTION
**Pull request by danny0405**
…dsException

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java [hunk 1]
```diff
@@ -414,6 +415,46 @@ protected DiffRepository getDiffRepos() {
         .check();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3171">[CALCITE-3171]
+   * SemiJoin on conditions push down throws IndexOutOfBoundsException</a>. */
+  @Test public void testPushSemiJoinConditionsToLeft() {
+    final RelBuilder relBuilder = RelBuilder.create(RelBuilderTest.config().build());
+    // build a rel equivalent to sql:
```
Comment (rubenada) [inline core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java]: minor errors in SQL description:
where emp.**deptno** 
in (select dept.deptno from dept where **emp**.empno **>** 20)

**Comment by rubenada** [comment]
LGTM

File: core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java (hunk 1 above)
Comment (danny0405) [reply core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java]: Yep, thanks for correcting.

Comment (rubenada) [reply core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java]: Sorry for being picky, there are still typos :)
where emp.dempno => where emp.deptno 
where emp.empno < 20 => where emp.empno > 20

Comment (danny0405) [reply core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java]: Oops, i'm really careless these days.

Comment (rubenada) [reply core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java]: No problem, thanks for fixing it!

**Pull request merged**